### ISSUE TITLE
Inconsistent error handling for CLI-git/libgit2

### ISSUE THREAD

**Issue by laggvar** [JuliaLang/Pkg.jl#3249]
I have a private registry that I can only reach while being at the office or using VPN from home.

When I'm not connected through VPN and want to add a package from the general registry, all registries will be updated. In case Pkg uses libgit2, it will notice that the private registry is not reachable and just skip it. But if Pkg uses CLI git, it appears `Pkg.add` just stops when it cannot connect to my private registry. So essentially, I need to connect to my private network through VPN just to add a public package. Easy to work around, but quite annoying.

**Comment by KristofferC** [comment]
I don't think this is related to error handling but just that julia is waiting for the `git` call to finish. Not much can be done about that since we are using your local git. Presumably there are some git settings you can change that tweaks how long it will try to access a repo before returning.
